src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/LoggerConfiguration.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host.Loggers
+{
+    /// <summary>
+    /// Provides a filter for use with an <see cref="ILogger"/>.
+    /// </summary>
+    [System.CLSCompliant(false)]
+    public class LogCategoryFilter
```

Comment:
Rather than a filter interface that is specific to category, is there a way to make this a general filter that has access to category and other properties of the log event "object"? I'm anticipating that in the future people may want to filter on something other than just category. Does the ILogger fx have a built in notion of filters?

E.g. if this filter implemented ILogger, it would receive all event properties and could filter as needed, skipping any logs as needed. I believe we've used a similar pattern with TraceWriter and binding providers.